dask/dataframe/groupby.py

```diff
@@ -1161,6 +1162,32 @@ def _aca_agg(
         token = self._token_prefix + token
         levels = _determine_levels(self.index)
 
+        first_index = self.index if not isinstance(self.index, list) else self.index[0]
+        embarrassingly_parallel = (
+            self.obj.known_divisions
+            and self.obj._contains_index_name(
+                first_index if not is_index_like(first_index) else first_index.name
+            )
+            and len(set(self.obj.divisions)) == len(self.obj.divisions)
```

Comment:
Could we warn or error in this case maybe? idk. If you're expecting that "groupby over a sorted index is embarrassingly parallel", which I think should be a guarantee of the API after this PR, silently falling back on the slow path for this rather obscure reason seems confusing. In most cases I'd imagine users would prefer to be informed, so they can go back and use unique divisions so they can get the fast groupby.

Also funnily enough, until https://github.com/dask/dask/issues/8437 is fixed, we're actually guaranteed that the same value is always in exactly one partition, so the `embarrassingly_parallel` path is actually safe right now because of this bug 😁

Reply:
lolol yeah I forgot about the last division always being a dupe. I don't think we should warn or error for this case unless the user has opted into `fast_path` or something that indicates that they want to do the groupby only if it can be done using this parallel approach. Otherwise existing workflows will fail.

Reply:
Fair. Probably depends on where we land with https://github.com/dask/dask/issues/8437 and https://github.com/dask/dask/pull/8393 and what gets in first. I do think a `PerformanceWarning` for now could be reasonable, since it wouldn't break existing workflows, but at least would give you a heads up.

Reply:
Ok that makes sense. I added a `PerformanceWarning`.